From 30656370d8418d469ed311fad84412d12fe0028e Mon Sep 17 00:00:00 2001
From: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
Date: Tue, 28 Jun 2022 17:04:52 +1000
Subject: [PATCH] xorg::proto: aero specific changes

Signed-off-by: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
---
 .gitignore          | 3 +++
 include/X11/Xos.h   | 2 +-
 include/X11/Xos_r.h | 2 +-
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index 1baa360..dc18465 100644
--- a/.gitignore
+++ b/.gitignore
@@ -18,3 +18,6 @@ man/Xprint.7
 missing
 stamp-h1
 stamp-h2
+
+# editor configs:
+.vscode
diff --git a/include/X11/Xos.h b/include/X11/Xos.h
index 28dfc67..1c4092e 100644
--- a/include/X11/Xos.h
+++ b/include/X11/Xos.h
@@ -60,7 +60,7 @@ in this Software without prior written authorization from The Open Group.
  */
 
 # include <string.h>
-# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__aero__)
 #  include <strings.h>
 # else
 #  ifndef index
diff --git a/include/X11/Xos_r.h b/include/X11/Xos_r.h
index f963b64..19ef5f4 100644
--- a/include/X11/Xos_r.h
+++ b/include/X11/Xos_r.h
@@ -318,7 +318,7 @@ static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
   (_Xos_processUnlock), \
   (p).pwp )
 
-#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__)
+#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__) && !defined(__aero__)
 # define X_NEEDS_PWPARAMS
 typedef struct {
   struct passwd pws;
-- 
2.25.1

