diff --git glib-clean/gio/gcredentialsprivate.h glib-workdir/gio/gcredentialsprivate.h
index 0310a75..f967f90 100644
--- glib-clean/gio/gcredentialsprivate.h
+++ glib-workdir/gio/gcredentialsprivate.h
@@ -104,7 +104,7 @@
  */
 #undef G_CREDENTIALS_HAS_PID
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__aero__)
 #define G_CREDENTIALS_SUPPORTED 1
 #define G_CREDENTIALS_USE_LINUX_UCRED 1
 #define G_CREDENTIALS_NATIVE_TYPE G_CREDENTIALS_TYPE_LINUX_UCRED
diff --git glib-clean/glib/giounix.c glib-workdir/glib/giounix.c
index 84e8135..5b68f25 100644
--- glib-clean/glib/giounix.c
+++ glib-workdir/glib/giounix.c
@@ -440,7 +440,10 @@ g_io_unix_get_flags (GIOChannel *channel)
         channel->is_writeable = TRUE;
         break;
       default:
-        g_assert_not_reached ();
+        channel->is_readable = TRUE;
+        channel->is_writeable = TRUE;
+        break;
+        // g_assert_not_reached ();
     }
 
   return flags;
diff --git glib-clean/glib/glib-init.c glib-workdir/glib/glib-init.c
index 933f891..393e51d 100644
--- glib-clean/glib/glib-init.c
+++ glib-workdir/glib/glib-init.c
@@ -29,6 +29,7 @@
 #include "gmem.h"       /* for g_mem_gc_friendly */
 
 #include <string.h>
+#include <strings.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
diff --git glib-clean/glib/gstrfuncs.c glib-workdir/glib/gstrfuncs.c
index 22a608d..29e123c 100644
--- glib-clean/glib/gstrfuncs.c
+++ glib-workdir/glib/gstrfuncs.c
@@ -35,6 +35,7 @@
 #include <stdlib.h>
 #include <locale.h>
 #include <string.h>
+#include <strings.h>
 #include <locale.h>
 #include <errno.h>
 #include <garray.h>
