From efd05717147288b099e4ad390871c379035cd1e1 Mon Sep 17 00:00:00 2001
From: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
Date: Thu, 17 Nov 2022 19:15:02 +1100
Subject: [PATCH] <xxx>

---
 xterm.c | 22 +++++++---------------
 1 file changed, 7 insertions(+), 15 deletions(-)

diff --git a/xterm.c b/xterm.c
index 51b5904..01ce17e 100644
--- a/xterm.c
+++ b/xterm.c
@@ -129,7 +129,6 @@ int main(int argc, char **argv) {
         close(pty_master);
         ioctl(pty_slave, TIOCSCTTY, 0);
         ioctl(pty_slave, TIOCSWINSZ, &win_size);
-        setsid();
 
         dup2(pty_slave, 0);
         dup2(pty_slave, 1);
@@ -139,11 +138,13 @@ int main(int argc, char **argv) {
         execlp("/bin/bash", "/bin/bash", "-l", NULL);
     }
 
-    pthread_t pty_thread;
+    // pthread_t pty_thread;
 
-    if (pthread_create(&pty_thread, NULL, read_from_pty, NULL) != 0) {
-        goto cleanup;
-    }
+    // if (pthread_create(&pty_thread, NULL, read_from_pty, NULL) != 0) {
+    //     goto cleanup;
+    // }
+
+    term_write(context, "Hi, from limine-term with X backend!", strlen("Hi, from limine-term with X backend!"));
 
     while (true) {
         XEvent e;
@@ -159,16 +160,7 @@ int main(int argc, char **argv) {
     XImage *image = XCreateImage(display, DefaultVisual(display, screen), depth, ZPixmap, 0, (char *)framebuffer, win_size.ws_xpixel, win_size.ws_ypixel, 32, 0);
 
     for (;;) {
-        while (XPending(display) > 0) {
-            XEvent e;
-            XNextEvent(display, &e);
-
-            if (e.type == KeyPress) {
-                // TODO
-            }
-
-            XPutImage(display, window, gc, image, 0, 0, 0, 0, win_size.ws_xpixel, win_size.ws_ypixel);
-        }
+        XPutImage(display, window, gc, image, 0, 0, 0, 0, win_size.ws_xpixel, win_size.ws_ypixel);
     }
 
 cleanup:
-- 
2.38.1

