diff --git a/bundled/coreutils-orig/build-aux/config.sub b/bundled/coreutils-workdir/build-aux/config.sub
index f02d43a..06e0dab 100755
--- a/bundled/coreutils-orig/build-aux/config.sub
+++ b/bundled/coreutils-workdir/build-aux/config.sub
@@ -1346,7 +1346,7 @@ case $os in
 	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
-	     | knetbsd* | mirbsd* | netbsd* \
+	     | knetbsd* | mirbsd* | netbsd* | aero* \
 	     | bitrig* | openbsd* | solidbsd* | libertybsd* | os108* \
 	     | ekkobsd* | kfreebsd* | freebsd* | riscix* | lynxos* \
 	     | bosx* | nextstep* | cxux* | aout* | elf* | oabi* \