From 634f15ee7db8317cb39e1f13dbd1e4cd16f922e6 Mon Sep 17 00:00:00 2001
From: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
Date: Wed, 25 May 2022 16:43:44 +1000
Subject: [PATCH] libtool: aero specific changes

Signed-off-by: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
---
 build-aux/ltmain.in |  2 +-
 m4/libtool.m4       | 15 +++++++++++++++
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index d5cf07a..ce97a5f 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -6449,7 +6449,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      add_dir=-L$lt_sysroot$libdir
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
diff --git a/m4/libtool.m4 b/m4/libtool.m4
index a3bc337..9136171 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -2905,6 +2905,17 @@ netbsd*)
   hardcode_into_libs=yes
   ;;
 
+aero*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 newsos6)
   version_type=linux # correct to gnu/linux during the next big refactor
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
@@ -3554,6 +3565,10 @@ netbsd*)
   fi
   ;;
 
+aero*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 newos6*)
   lt_cv_deplibs_check_method='file_magic ELF [[0-9]][[0-9]]*-bit [[ML]]SB (executable|dynamic lib)'
   lt_cv_file_magic_cmd=/usr/bin/file
-- 
2.25.1

