From 8c7f96493b8706d43b30f49d56e0e84ee27fe65e Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 27 Apr 2022 20:27:21 +0200
Subject: [PATCH 1/2] Add a missing include

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 lib/dbm/src/h_page.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/dbm/src/h_page.c b/lib/dbm/src/h_page.c
index d4e4ff6..7b3271e 100644
--- a/lib/dbm/src/h_page.c
+++ b/lib/dbm/src/h_page.c
@@ -81,6 +81,7 @@ static char sccsid[] = "@(#)hash_page.c 8.7 (Berkeley) 8/16/94";
 #endif
 
 #include <assert.h>
+#include <endian.h>
 
 #include "mcom_db.h"
 #include "hash.h"
-- 
2.36.0
