From 006807c2ec4429094c7bc0b1f54a7db462916df1 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 14 Jul 2021 02:35:21 +0200
Subject: [PATCH] Remove test directory as that can't be cross-compiled

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 86cca65..76b49b1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,7 +27,7 @@ EXTRA_DIST = windows/asn1-parser/asn1-parser.vcproj			\
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk
 EXTRA_DIST += cfg.mk maint.mk CONTRIBUTING.md README.md
 
-SUBDIRS = lib src fuzz tests
+SUBDIRS = lib src fuzz
 
 if ENABLE_DOC
 SUBDIRS += doc examples
-- 
2.32.0
