From f5fab150ee8fbfdef46888837b6e6d43fd3ec1c0 Mon Sep 17 00:00:00 2001
From: Andy-Python-Programmer <andypythonappdeveloper@gmail.com>
Date: Sun, 20 Nov 2022 13:39:51 +1100
Subject: [PATCH] <xxx>

---
 src/os/unix/consts.rs | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/os/unix/consts.rs b/src/os/unix/consts.rs
index dbe4df9..fbdd2e3 100644
--- a/src/os/unix/consts.rs
+++ b/src/os/unix/consts.rs
@@ -58,7 +58,7 @@ mod posix {
     use self::cfg_if::cfg_if;
     use super::c_int;
     cfg_if! {
-        if #[cfg(target_os = "haiku")] {
+        if #[cfg(any(target_os = "haiku", target_os = "aero"))] {
             pub(super) const RTLD_LAZY: c_int = 0;
         } else if #[cfg(any(
             target_os = "linux",
@@ -90,7 +90,7 @@ mod posix {
     }
 
     cfg_if! {
-        if #[cfg(target_os = "haiku")] {
+        if #[cfg(any(target_os = "haiku", target_os = "aero"))] {
             pub(super) const RTLD_NOW: c_int = 1;
         } else if #[cfg(any(
             target_os = "linux",
@@ -126,6 +126,7 @@ mod posix {
     cfg_if! {
         if #[cfg(any(
             target_os = "haiku",
+            target_os = "aero",
             all(target_os = "android",target_pointer_width = "32"),
         ))] {
             pub(super) const RTLD_GLOBAL: c_int = 2;
@@ -193,6 +194,7 @@ mod posix {
 
             target_os = "fuchsia",
             target_os = "redox",
+            target_os = "aero",
         ))] {
             pub(super) const RTLD_LOCAL: c_int = 0;
         } else {
-- 
2.38.1

