From d6dfe23ee9299412d8281dabb7f5d903e152b589 Mon Sep 17 00:00:00 2001
From: Anhad Singh <andypythonappdeveloper@gmail.com>
Date: Tue, 10 Oct 2023 17:24:17 +1100
Subject: [PATCH] <xxx>

Signed-off-by: Anhad Singh <andypythonappdeveloper@gmail.com>
---
 src/os/unix/consts.rs | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/os/unix/consts.rs b/src/os/unix/consts.rs
index ed3edaf..61659ff 100644
--- a/src/os/unix/consts.rs
+++ b/src/os/unix/consts.rs
@@ -58,7 +58,7 @@ mod posix {
     use self::cfg_if::cfg_if;
     use super::c_int;
     cfg_if! {
-        if #[cfg(target_os = "haiku")] {
+        if #[cfg(any(target_os = "haiku", target_os = "aero"))] {
             pub(super) const RTLD_LAZY: c_int = 0;
         } else if #[cfg(target_os = "aix")] {
             pub(super) const RTLD_LAZY: c_int = 4;
@@ -94,7 +94,7 @@ mod posix {
     }
 
     cfg_if! {
-        if #[cfg(target_os = "haiku")] {
+        if #[cfg(any(target_os = "haiku", target_os = "aero"))] {
             pub(super) const RTLD_NOW: c_int = 1;
         } else if #[cfg(any(
             target_os = "linux",
@@ -133,6 +133,7 @@ mod posix {
     cfg_if! {
         if #[cfg(any(
             target_os = "haiku",
+            target_os = "aero",
             all(target_os = "android",target_pointer_width = "32"),
         ))] {
             pub(super) const RTLD_GLOBAL: c_int = 2;
@@ -210,6 +211,7 @@ mod posix {
             target_os = "fuchsia",
             target_os = "redox",
             target_os = "hurd",
+            target_os = "aero",
         ))] {
             pub(super) const RTLD_LOCAL: c_int = 0;
         } else {
-- 
2.42.0

